### PR TITLE
add Timelines::Event

### PR DESCRIPTION
**Pull request by metamoni**
Adds a `Timelines::Event` model and a generator, to allow users to run `bin/rails generate timelines:install` in order to add a migration.

This PR also removes the unused `railtie.rb` file, as discussed [here](https://app.shortcut.com/devlin-mambo-llp/story/1263/update-the-timelines-ephemeral-concern-to-offer-an-active-at-scope-and-active-at-instance-method).

### CODE REVIEW

File: lib/timelines/models/concerns/has_events.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require "active_support/concern"
+
+module Timelines
```
Comment (metamoni) [inline lib/timelines/models/concerns/has_events.rb:3]: Have I put this in the right folder? Does it need to be inside `/models`? That's how it was structured in the main insights repo.

File: lib/timelines/models/concerns/has_events.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  module HasEvents
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :events
```
Comment (metamoni) [inline lib/timelines/models/concerns/has_events.rb]: Not used concerns before. What are we looking to achieve here, how do I test that I've set this up properly in the Rails console?

File: lib/generators/timelines/install/templates/create_timelines_events.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreatePermitsInvites < ActiveRecord::Migration[7.1]
```
Comment (Craggar) [inline lib/generators/timelines/install/templates/create_timelines_events.rb]: This should be `class CreateTimelinesEvents...`

File: lib/generators/timelines/install/templates/create_timelines_events.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  def change
+    create_table :permits_events, id: :uuid do |t|
```
Comment (Craggar) [inline lib/generators/timelines/install/templates/create_timelines_events.rb]: the table name here should be `timelines_events`

File: lib/timelines.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 require "timelines/version"
 require "timelines/railtie"
-require "timelines/ephemeral"
+require "timelines/models/concerns/ephemeral"
+require "timelines/models/event"
```
Comment (Craggar) [inline lib/timelines.rb]: You also need to require your new concern `require "timelines/models/concerns/has_events"`

File: lib/timelines/models/concerns/has_events.rb (hunk 1 above)
Comment (Craggar) [reply lib/timelines/models/concerns/has_events.rb:3]: Yep, all good 👍

File: lib/timelines/models/concerns/has_events.rb (hunk 2 above)
Comment (Craggar) [reply lib/timelines/models/concerns/has_events.rb]: This looks good, but you'll need to specify a class name for the has_many: 
`has_many :events, class_name: "Timelines::Event", as: :resource, dependent: :destroy`

File: lib/timelines/models/event.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+module Timelines
+  class Event < ActiveRecord::Base
+    belongs_to :actor, polymorphic: true
```
Comment (Craggar) [inline lib/timelines/models/event.rb:3]: We'll need to specify `self.table_name = "timelines_events"` here

**Review by Craggar — CHANGES_REQUESTED** [review]
This is looking really good so far. I've added a few change requests which I came across as a result of attempting adding RSpec to the project so we can start getting some test coverage.
Once you've made those changes, and we have this merged, I can add a meaningful rspec test suite over it, but that's ready to go once this PR is done.

File: Gemfile.lock [hunk 7]
```diff
@@ -0,0 +1,203 @@
+PATH
```
Comment (metamoni) [inline Gemfile.lock]: Not sure if we need to include this file?

Comment (Craggar) [reply Gemfile.lock]: No, don't think it's needed (it'll get populated when I add RSpec)

**Review by Craggar — APPROVED** [review]

**Pull request merged**